vllm/spec_decode/spec_decode_worker.py

```diff
@@ -143,15 +170,16 @@ def __init__(
         self.proposer_worker = proposer_worker
         self.scorer_worker = scorer_worker
         self.disable_by_batch_size = disable_by_batch_size or float("inf")
-        self.rejection_sampler = rejection_sampler
-
+        self.spec_decode_sampler = spec_decode_sampler
+        assert (
+            self.spec_decode_sampler is not None,
+            "Sampler is Not set, which is not expected."
+        )
```

Comment:
nit: we should rely on upper layers to provide validation and raise; we don't need to assert something basic this deep.

Reply:
Done. Removed this here. Added validations earlier.